### DISCUSSION TITLE
OTLP->Zipkin remoteEndPoint mapping question

### DISCUSSION THREAD

**Discussion by evantorrie** [open-telemetry/opentelemetry-specification#1546] [Q&A]
The [current specification](https://github.com/open-telemetry/opentelemetry-specification/blob/cb16422a61219d4fd99b466a70e47cb8af9e26b1/specification/trace/sdk_exporters/zipkin.md#otlp---zipkin) states that 

> If Zipkin `SpanKind` resolves to either `SpanKind.CLIENT` or `SpanKind.PRODUCER`
then the service SHOULD specify remote endpoint otherwise Zipkin won't treat the
Span as a dependency.

This seems like a typo to me regarding `SpanKind.CLIENT`. Shouldn't that be `SpanKind.CONSUMER` based on [these comments in the Zipkin](https://github.com/openzipkin/zipkin/blob/eeb17681af780e563b44503d5af1eadb3ca6abfc/zipkin/src/main/java/zipkin2/Span.java#L107-L129) repository?

**Comment by jcchavezs** [comment]
According to the documentation here https://github.com/openzipkin/brave/blob/0a5666d4a918b2a359cefff84e276bc9bbffa5a4/brave/src/main/java/brave/Span.java#L50

     /**
     * When present, {@link #start()} is the moment a producer sent a message to a destination. A
     * duration between {@link #start()} and {@link #finish()} may imply batching delay. {@link
     * #remoteEndpoint(Endpoint)} indicates the destination, such as a broker.
     *
     * <p>Unlike {@link #CLIENT}, messaging spans never share a span ID. For example, the {@link
     * #CONSUMER} of the same message has {@link TraceContext#parentId()} set to this span's {@link
     * TraceContext#spanId()}.
     */
    PRODUCER,
    /**
     * When present, {@link #start()} is the moment a consumer received a message from an origin. A
     * duration between {@link #start()} and {@link #finish()} may imply a processing backlog. while
     * {@link #remoteEndpoint(Endpoint)} indicates the origin, such as a broker.
     *
     * <p>Unlike {@link #SERVER}, messaging spans never share a span ID. For example, the {@link
     * #PRODUCER} of this message is the {@link TraceContext#parentId()} of this span.
     */
    CONSUMER
    
Meaning both producer and consumer should have it. In the other hand, CLIENT has always used remoteEndpoint so I'd say all three should have it. Any input @anuraaga ?

**Reply by anuraaga** [reply]
I think we can update the doc to just not mention `SpanKind` - we should set it when we have it available, regardless of the type. The extra documentation in the zipkin spec is just to clarify the remote endpoint is the broker, not the other side actually prducing/processing the messages. In all kinds, there are still cases we wouldn't be able to set it, such as unix domain sockets.

**Reply by evantorrie** [reply]
I opened #1560 as an issue to change this wording.
